glambie/util/timeseries_helpers.py

```diff
@@ -593,3 +596,115 @@ def get_average_trends_over_new_time_periods(start_dates, end_dates, changes, ne
+            new_errors.append(np.nan)
+            new_hydro.append(np.nan)
+            new_sea_level.append(np.nan)
+
+    # Add last row in
+    new_start_dates.append(start_dates[-1])
+    new_end_dates.append(end_dates[-1])
+    new_changes.append(input_dataframe.glacier_change_observed.tolist()[-1])
+    new_errors.append(input_dataframe.glacier_change_uncertainty.tolist()[-1])
+    new_hydro.append(input_dataframe.hydrological_correction_value.tolist()[-1])
+    new_sea_level.append(input_dataframe.sea_level_correction_value.tolist()[-1])
+
+    interpolated_dataframe = pd.DataFrame()
+    interpolated_dataframe['start_date'] = [datetime.datetime.strftime(a, '%d/%m/%Y') for a in new_start_dates]
+    interpolated_dataframe['end_date'] = [datetime.datetime.strftime(a, '%d/%m/%Y') for a in new_end_dates]
+    interpolated_dataframe['glacier_change_observed'] = new_changes
+    interpolated_dataframe['glacier_change_uncertainty'] = new_errors
+    interpolated_dataframe['hydrological_correction_value'] = new_hydro
+    interpolated_dataframe['sea_level_correction_value'] = new_sea_level
+
+    # calculate days covered by each row
+    date_gaps = [
+        datetime.datetime.strptime(interpolated_dataframe['end_date'][i], '%d/%m/%Y') - datetime.datetime.strptime(
+            interpolated_dataframe['start_date'][i], '%d/%m/%Y') for i in range(len(interpolated_dataframe))]
+    date_gaps_in_days = [a.days for a in date_gaps]
+    interpolated_dataframe['days_covered'] = date_gaps_in_days
+
+    # calculate change per day in each row
+    interpolated_dataframe['glacier_change_per_day'] = [
+        a / b for a, b in zip(interpolated_dataframe.glacier_change_observed, interpolated_dataframe.days_covered)]
+    interpolated_dataframe['glacier_change_uncertainty_per_day'] = [
+        a / b for a, b in zip(interpolated_dataframe.glacier_change_uncertainty, interpolated_dataframe.days_covered)]
+
+    dates_list_fractional = datetime_dates_to_fractional_years(new_start_dates)
+    interpolated_dataframe['date_fractional'] = dates_list_fractional
+
+    # Linear interpolation of glacier_change_per_day to fill gaps
+    df = pd.DataFrame({'time': interpolated_dataframe['date_fractional'],
+                       'mass': interpolated_dataframe['glacier_change_per_day'],
+                       'error': interpolated_dataframe['glacier_change_uncertainty_per_day'],
+                       'hydro': interpolated_dataframe['hydrological_correction_value'],
+                       'sea_level': interpolated_dataframe['sea_level_correction_value']})
+    df_int = df.interpolate(method='linear')
+
+    interpolated_dataframe['glacier_change_per_day'] = df_int['mass'].values.tolist()
+    interpolated_dataframe['glacier_change_uncertainty_per_day'] = df_int['error'].values.tolist()
+    interpolated_dataframe['hydrological_correction_value'] = df_int['hydro']
+    interpolated_dataframe['sea_level_correction_value'] = df_int['sea_level']
```

Comment:
could this be simplified to:
```suggestion
   for column_name in ['glacier_change_per_day', 'glacier_change_uncertainty_per_day', 'hydrological_correction_value', 'sea_level_correction_value']:
       interpolated_dataframe[column_name] = interpolated_dataframe[column_name].interpolate(method='linear')
```

Reply:
Yes, much simpler, thanks!